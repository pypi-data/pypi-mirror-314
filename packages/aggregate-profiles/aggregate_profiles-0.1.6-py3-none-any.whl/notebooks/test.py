import aggregate_profiles as aggp 

print(dir(aggp))