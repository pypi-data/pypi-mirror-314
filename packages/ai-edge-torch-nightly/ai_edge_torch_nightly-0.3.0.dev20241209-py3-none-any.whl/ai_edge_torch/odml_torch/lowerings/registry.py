# Copyright 2024 The AI Edge Torch Authors.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
# ==============================================================================
"""Torch op decompositions and MLIR lowerings registry."""

from typing import Any, Callable

import torch

from . import context


class LoweringRegistry:
  """Registry object for torch op decompositions and to-MLIR lowerings."""

  def __init__(self):
    self.registered_ops = {}
    self.decompositions = {}

  def lookup(self, op_or_name):
    candidate = self._get_lowering(op_or_name)
    if candidate is None:
      if isinstance(op_or_name, torch._ops.OpOverloadPacket):
        candidate = self._get_lowering(op_or_name.default)
      if isinstance(op_or_name, torch._ops.OpOverload):
        candidate = self._get_lowering(op_or_name.overloadpacket)
    return candidate

  def _get_lowering(self, op):
    candidate = self.registered_ops.get(op)
    return candidate

  def register(self, op, lowering):
    if isinstance(op, torch._ops.OpOverloadPacket):
      ops = [getattr(op, overload) for overload in op.overloads()]
    else:
      ops = [op]

    for op in ops:
      self.registered_ops[op] = lowering


global_registry = LoweringRegistry()
global_registry.decompositions.update(torch._decomp.core_aten_decompositions())
global_registry.decompositions.update(
    torch._decomp.get_decompositions([
        torch.ops.aten.upsample_nearest2d,
        torch.ops.aten._native_batch_norm_legit.no_stats,
        torch.ops.aten._native_batch_norm_legit_functional,
        torch.ops.aten._adaptive_avg_pool2d,
        torch.ops.aten._adaptive_avg_pool3d,
        torch.ops.aten.grid_sampler_2d,
        torch.ops.aten.native_group_norm,
        torch.ops.aten.native_dropout,
        torch.ops.aten.reflection_pad1d,
        torch.ops.aten.reflection_pad2d,
        torch.ops.aten.reflection_pad3d,
        torch.ops.aten.replication_pad1d,
        torch.ops.aten.replication_pad2d,
        torch.ops.aten.replication_pad3d,
        torch.ops.aten.addmm,
    ])
)

torch._decomp.remove_decompositions(
    global_registry.decompositions,
    [
        torch.ops.aten.roll,
    ],
)


def lookup(op):
  return global_registry.lookup(op)


def lower(op):
  def inner(lowering: Callable[[context.LoweringContext, ...], Any]):
    global_registry.register(op, lowering)
    return lowering

  return inner


def decompositions():
  return global_registry.decompositions
