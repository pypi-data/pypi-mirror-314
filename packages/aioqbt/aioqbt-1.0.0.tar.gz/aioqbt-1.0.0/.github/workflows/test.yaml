---
name: test

on:
  push:

jobs:
  test:
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
      matrix:
        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11", "3.12", "3.13"]
    steps:
      - name: Checkout
        uses: actions/checkout@v3

      - name: Set up Python ${{ matrix.python-version }}
        uses: actions/setup-python@v4
        with:
          python-version: ${{ matrix.python-version }}

      - name: Install qbittorrent-nox
        run: |
          sudo apt-get install --yes --no-install-recommends qbittorrent-nox

      - name: Install package
        run: |
          python -m pip install --upgrade pip
          python -m pip install .[test]

      - name: Run tests
        run: pytest
