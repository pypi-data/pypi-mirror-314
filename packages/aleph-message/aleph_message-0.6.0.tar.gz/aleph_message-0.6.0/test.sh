#!/bin/sh

set -euf

podman build --loglevel 3 -t aleph-message .
podman run --rm -ti -v $(pwd)/aleph_message:/opt/aleph_message aleph-message pytest -vv
podman run --rm -ti -v $(pwd)/aleph_message:/opt/aleph_message aleph-message mypy --ignore-missing-imports aleph_message
