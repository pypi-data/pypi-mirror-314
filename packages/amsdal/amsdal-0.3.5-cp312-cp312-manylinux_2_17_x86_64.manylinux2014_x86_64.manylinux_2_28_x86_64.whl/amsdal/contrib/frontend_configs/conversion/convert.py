from datetime import date
from datetime import datetime
from enum import Enum
from inspect import _empty
from inspect import signature
from types import FunctionType
from types import UnionType
from typing import Any
from typing import ClassVar
from typing import ForwardRef
from typing import Union

from amsdal_models.classes.manager import ClassManager
from amsdal_models.classes.model import LegacyModel
from amsdal_models.classes.model import Model
from amsdal_utils.models.data_models.reference import Reference
from pydantic import BaseModel
from pydantic_core import PydanticUndefined

from amsdal.schemas.manager import SchemaManager

default_types_map = {
    int: 'number',
    float: 'number',
    bool: 'checkbox',
    str: 'text',
    bytes: 'Bytes',
    date: 'date',
    datetime: 'datetime',
}


def _process_union(value: UnionType, *, is_transaction: bool = False) -> dict[str, Any]:
    arg_type = {'required': True}
    for arg in value.__args__:
        if arg is type(None):
            arg_type['required'] = False
            continue

        control = convert_to_frontend_config(arg, is_transaction=is_transaction)
        if control:
            arg_type.update(control)

    return arg_type


def convert_to_frontend_config(value: Any, *, is_transaction: bool = False) -> dict[str, Any]:
    """
    Converts a given value to a frontend configuration dictionary.

    This function takes a value and converts it into a dictionary that represents
    the configuration for a frontend form control. It handles various types such as
    Union, list, dict, BaseModel, and custom types.

    Args:
        value (Any): The value to be converted to frontend configuration.
        is_transaction (bool, optional): Indicates if the conversion is for a transaction. Defaults to False.

    Returns:
        dict[str, Any]: A dictionary representing the frontend configuration for the given value.
    """
    if hasattr(value, '__origin__'):
        origin_class = value.__origin__

        if origin_class in [ClassVar]:
            return {}

        if origin_class is Union:
            _union = _process_union(value, is_transaction=is_transaction)

            if 'entityType' in _union and _union['entityType'] == 'File':
                _union['type'] = 'file'
                del _union['entityType']

            return _union

        if origin_class is list:
            return {
                'type': 'array',
                'name': 'array_items',
                'label': 'array_items',
                'control': {
                    'name': 'array_items_values',
                    'label': 'array_items_values',
                    **convert_to_frontend_config(value.__args__[0], is_transaction=is_transaction),
                },
            }

        if origin_class is dict:
            return {
                'type': 'dict',
                'name': 'dict_items',
                'label': 'dict_items',
                'control': {
                    'name': 'dict_items_values',
                    'label': 'dict_items_values',
                    **convert_to_frontend_config(value.__args__[1], is_transaction=is_transaction),
                },
            }

    if isinstance(value, ForwardRef):
        class_name = value.__forward_arg__
        _class = ClassManager().import_class(class_name, ClassManager().resolve_schema_type(class_name))

        if issubclass(_class, Model):
            return {
                'entityType': value.__forward_arg__,
            }

        value = _class

    if isinstance(value, UnionType):
        _union = _process_union(value, is_transaction=is_transaction)

        if 'entityType' in _union and _union['entityType'] == 'File':
            _union['type'] = 'file'
            del _union['entityType']

        return _union

    if value in default_types_map:
        return {
            'type': default_types_map[value],
        }

    if value is Any:
        return {
            'type': 'text',
        }

    if isinstance(value, FunctionType):
        function_controls = []

        while hasattr(value, '__wrapped__'):
            value = value.__wrapped__

        _signature = signature(value)
        _parameters = _signature.parameters

        for arg_name in _parameters:
            if arg_name in value.__annotations__:
                arg_type = value.__annotations__[arg_name]
                control = convert_to_frontend_config(
                    arg_type,
                    is_transaction=True,
                )

                if not control:
                    continue

                control['name'] = arg_name
                control['label'] = arg_name
            else:
                control = {
                    'type': 'text',
                    'name': arg_name,
                    'label': arg_name,
                }

            control.setdefault('required', True)
            _param = _parameters[arg_name]
            if _param.default is not _empty:
                control['value'] = _param.default
                control['required'] = False

            function_controls.append(control)

        return {
            'type': 'group',
            'name': value.__name__,
            'label': value.__name__,
            'controls': function_controls,
        }

    if issubclass(value, Reference):
        return {
            'type': 'object_latest',
        }

    if is_transaction and issubclass(value, Model):
        if value.__name__ == 'File':
            return {
                'type': 'file',
            }
        return {
            'type': 'object_latest',
            'entityType': value.__name__,
        }

    if issubclass(value, LegacyModel):
        return {}

    if issubclass(value, BaseModel):
        model_controls = []

        try:
            schema = SchemaManager().get_schema_by_name(value.__name__)
        except FileNotFoundError:
            schema = None

        for field_name, field in value.__annotations__.items():
            control = convert_to_frontend_config(field, is_transaction=is_transaction)

            if not control:
                continue

            control.setdefault('required', True)

            if field_name in value.model_fields:
                _field = value.model_fields[field_name]

                if _field.default is not PydanticUndefined:
                    control['value'] = _field.default

            control['name'] = field_name
            control['label'] = field_name

            if schema and schema.properties and field_name in schema.properties:
                schema_property = schema.properties[field_name]

                if schema_property.options:
                    control['options'] = [
                        {
                            'label': option.key,
                            'value': option.value,
                        }
                        for option in schema_property.options
                    ]
                    if control.get('type') == 'text':
                        control['type'] = 'select'

                if schema_property.title:
                    control['label'] = schema_property.title

            model_controls.append(control)

        return {
            'type': 'group',
            'name': value.__name__,
            'label': value.__name__,
            'controls': model_controls,
        }

    if issubclass(value, Enum):
        return {
            'type': 'select',
            'options': [{'label': option.name, 'value': option.value} for option in value],
        }

    return {}
