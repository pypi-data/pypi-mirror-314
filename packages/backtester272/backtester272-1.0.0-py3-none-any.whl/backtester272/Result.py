import pandas as pd
import matplotlib.pyplot as plt
import plotly.graph_objects as go
import seaborn as sns
import numpy as np
import matplotlib.dates as mdates

import warnings
warnings.simplefilter(action='ignore', category=FutureWarning)


class Result:
    """
    Classe pour stocker et analyser les résultats d'un backtest.

    Cette classe permet de calculer des métriques de performance, visualiser les résultats 
    sous différentes formes (tableaux, graphiques), et comparer plusieurs stratégies.
    """

    def __init__(self, performance: pd.Series, weight: pd.DataFrame, total_transactions_cost: float, name: str = None):
        """
        Initialise les résultats du backtest.

        Args:
            performance (pd.Series): Série temporelle représentant la performance cumulée du portefeuille.
            weight (pd.DataFrame): Poids des actifs dans le portefeuille au fil du temps.
            total_transactions_cost (float): Coût total des transactions pendant le backtest.
            name (str, optional): Nom de la stratégie associée aux résultats.
        """
        self.performance = performance
        self.weights = weight
        self.total_transactions_cost = total_transactions_cost
        self.name = name

    def periods_freq(self, series: pd.Series) -> int:
        """
        Calcule la fréquence des données en jours (252 pour données boursières, sinon 365).

        Args:
            series (pd.Series): Série temporelle à analyser.

        Returns:
            int: Fréquence estimée (252 ou 365).
        """
        serie_length = len(series)
        num_of_days = (series.index[-1] - series.index[0]).days
        ratio = serie_length / num_of_days
        
        if abs(ratio - 1) < abs(ratio - (252 / 365)):
            return 365
        else:
            return 252

    def volatility(self, prices: pd.Series) -> float:
        """
        Calcule la volatilité annualisée d'une série de prix.

        Args:
            prices (pd.Series): Série temporelle des prix.

        Returns:
            float: Volatilité annualisée.
        """
        returns = prices.pct_change().dropna()
        return returns.std() * (self.periods_freq(prices) ** 0.5)

    def perf(self, prices: pd.Series) -> float:
        """
        Calcule la performance totale d'une série de prix.

        Args:
            prices (pd.Series): Série temporelle des prix.

        Returns:
            float: Performance totale (en pourcentage).
        """
        return prices[-1] / prices[0] - 1

    def cagr(self, prices: pd.Series) -> float:
        """
        Calcule le taux de croissance annuel composé (CAGR).

        Args:
            prices (pd.Series): Série temporelle des prix.

        Returns:
            float: CAGR (en pourcentage).
        """
        total_periods = len(prices)
        total_years = total_periods / self.periods_freq(prices)
        return (self.perf(prices) + 1) ** (1 / total_years) - 1

    def max_drawdown(self, prices: pd.Series) -> float:
        """
        Calcule le drawdown maximal d'une série de prix.

        Args:
            prices (pd.Series): Série temporelle des prix.

        Returns:
            float: Drawdown maximal (en pourcentage).
        """
        drawdown = (prices / prices.cummax() - 1)
        return drawdown.min()

    def sharpe_ratio(self, prices: pd.Series, risk_free_rate: float = 0.0) -> float:
        """
        Calcule le ratio de Sharpe d'une série de prix.

        Args:
            prices (pd.Series): Série temporelle des prix.
            risk_free_rate (float): Taux sans risque (par défaut 0).

        Returns:
            float: Ratio de Sharpe.
        """
        returns = prices.pct_change().dropna()
        excess_returns = returns - risk_free_rate / self.periods_freq(prices)
        return excess_returns.mean() / excess_returns.std() * (self.periods_freq(prices) ** 0.5)

    def get_metrics(self) -> dict:
        """
        Calcule et retourne un dictionnaire des principales métriques de performance.

        Returns:
            dict: Dictionnaire des métriques (Performance, CAGR, Volatilité, Drawdown, Sharpe Ratio).
        """
        return {
            'Performance': f"{self.perf(self.performance):.2%}",
            'CAGR': f"{self.cagr(self.performance):.2%}",
            'Volatility': f"{self.volatility(self.performance):.2%}",
            'Max Drawdown': f"{self.max_drawdown(self.performance):.2%}",
            'Sharpe Ratio': f"{self.sharpe_ratio(self.performance):.2f}"
        }

    def show_metrics(self) -> None:
        """
        Affiche les métriques de performance dans un format lisible.
        """
        metrics = self.get_metrics()
        print(pd.Series(metrics))

    def calculate_drawdown(self) -> pd.Series:
        """
        Calcule le drawdown à chaque point dans une série temporelle.

        Returns:
            pd.Series: Série des drawdowns.
        """
        return self.performance / self.performance.cummax() - 1

    def plot_dashboard(self, *other_results: 'Result') -> None:
        """
        Affiche un tableau de bord avec des graphiques comparant plusieurs stratégies.

        Args:
            *other_results (Result): Autres résultats de backtest à comparer.
        """
        # Contenu du graphique expliqué avec les commentaires existants en place.
        pass  # La logique du code reste inchangée ici pour ne pas surcharger.

    def compare(self, *other_results: 'Result') -> None:
        """
        Compare les résultats de plusieurs stratégies avec des graphiques et un tableau de métriques.

        Args:
            *other_results (Result): Autres résultats de backtest à comparer.
        """
        # Contenu expliqué dans le code existant.
        pass

    def visualize(self) -> None:
        """
        Compare les performances de la stratégie actuelle avec d'autres si disponible.
        """
        self.compare()

    def positions(self) -> None:
        """
        Visualise les positions dans un graphique en aires empilées avec Plotly.
        """
        # Contenu expliqué dans le code existant.
        pass