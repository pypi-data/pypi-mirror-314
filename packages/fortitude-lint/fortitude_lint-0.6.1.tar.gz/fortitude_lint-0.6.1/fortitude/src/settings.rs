use ruff_diagnostics::Applicability;
use ruff_macros::CacheKey;
use serde::{Deserialize, Serialize};
/// A collection of user-modifiable settings. Should be expanded as new features are added.
use std::fmt::{Display, Formatter};

use crate::rule_selector::RuleSelector;

pub struct Settings {
    pub line_length: usize,
}

impl Default for Settings {
    fn default() -> Self {
        Self { line_length: 100 }
    }
}

/// Toggle for rules still in preview
#[derive(Debug, Clone, Copy, PartialEq, Eq, Default, CacheKey, is_macro::Is)]
pub enum PreviewMode {
    #[default]
    Disabled,
    Enabled,
}

impl From<bool> for PreviewMode {
    fn from(version: bool) -> Self {
        if version {
            PreviewMode::Enabled
        } else {
            PreviewMode::Disabled
        }
    }
}

impl Display for PreviewMode {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        match self {
            Self::Disabled => write!(f, "disabled"),
            Self::Enabled => write!(f, "enabled"),
        }
    }
}

/// Default rule selection
pub const DEFAULT_SELECTORS: &[RuleSelector] = &[RuleSelector::All];

/// Toggle for unsafe fixes.
/// `Hint` will not apply unsafe fixes but a message will be shown when they are available.
/// `Disabled` will not apply unsafe fixes or show a message.
/// `Enabled` will apply unsafe fixes.
#[derive(Debug, Copy, Clone, CacheKey, Default, PartialEq, Eq, is_macro::Is)]
pub enum UnsafeFixes {
    #[default]
    Hint,
    Disabled,
    Enabled,
}

impl Display for UnsafeFixes {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        write!(
            f,
            "{}",
            match self {
                Self::Hint => "hint",
                Self::Disabled => "disabled",
                Self::Enabled => "enabled",
            }
        )
    }
}

impl From<bool> for UnsafeFixes {
    fn from(value: bool) -> Self {
        if value {
            UnsafeFixes::Enabled
        } else {
            UnsafeFixes::Disabled
        }
    }
}

impl UnsafeFixes {
    pub fn required_applicability(&self) -> Applicability {
        match self {
            Self::Enabled => Applicability::Unsafe,
            Self::Disabled | Self::Hint => Applicability::Safe,
        }
    }
}

/// Toggle for progress bar
#[derive(
    Clone, Copy, PartialEq, Eq, Serialize, Deserialize, Debug, Hash, Default, clap::ValueEnum,
)]
#[serde(rename_all = "kebab-case")]
pub enum ProgressBar {
    #[default]
    Off,
    Fancy,
    Ascii,
}

impl Display for ProgressBar {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        write!(
            f,
            "{}",
            match self {
                Self::Off => "off",
                Self::Fancy => "fancy",
                Self::Ascii => "ascii",
            }
        )
    }
}

#[derive(
    Clone, Copy, PartialEq, Eq, Serialize, Deserialize, Debug, Hash, Default, clap::ValueEnum,
)]
#[serde(rename_all = "kebab-case")]
pub enum OutputFormat {
    Concise,
    #[default]
    Full,
    Json,
    JsonLines,
    Junit,
    Grouped,
    Github,
    Gitlab,
    Pylint,
    Rdjson,
    Azure,
    Sarif,
}

impl Display for OutputFormat {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        match self {
            Self::Concise => write!(f, "concise"),
            Self::Full => write!(f, "full"),
            Self::Json => write!(f, "json"),
            Self::JsonLines => write!(f, "json_lines"),
            Self::Junit => write!(f, "junit"),
            Self::Grouped => write!(f, "grouped"),
            Self::Github => write!(f, "github"),
            Self::Gitlab => write!(f, "gitlab"),
            Self::Pylint => write!(f, "pylint"),
            Self::Rdjson => write!(f, "rdjson"),
            Self::Azure => write!(f, "azure"),
            Self::Sarif => write!(f, "sarif"),
        }
    }
}

#[derive(Debug, Copy, Clone, Hash, is_macro::Is)]
pub enum FixMode {
    Generate,
    Apply,
    #[allow(dead_code)]
    Diff,
}
