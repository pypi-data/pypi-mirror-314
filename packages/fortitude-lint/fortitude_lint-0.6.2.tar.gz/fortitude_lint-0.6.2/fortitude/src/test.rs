use std::path::Path;

use anyhow::Result;
use ruff_source_file::{SourceFile, SourceFileBuilder};

use crate::{
    check::{
        ast_entrypoint_map, check_file, read_to_string, rules_to_path_rules, rules_to_text_rules,
    },
    message::{Emitter, TextEmitter},
    rules::Rule,
    settings::{FixMode, Settings, UnsafeFixes},
};

pub(crate) fn test_resource_path(path: impl AsRef<Path>) -> std::path::PathBuf {
    Path::new("./resources/test/").join(path)
}

/// Run [`check_path`] on a Fortran file in the `resources/test/fixtures` directory.
pub(crate) fn test_path(
    path: impl AsRef<Path>,
    rules: &[Rule],
    settings: &Settings,
) -> Result<String> {
    let path = test_resource_path("fixtures").join(path);
    let source = read_to_string(&path)?;
    let filename = path.to_string_lossy();
    let file = SourceFileBuilder::new(filename.as_ref(), source.as_str()).finish();
    Ok(test_contents(&path, &file, rules, settings))
}

pub(crate) fn test_contents(
    path: &Path,
    file: &SourceFile,
    rules: &[Rule],
    settings: &Settings,
) -> String {
    let path_rules = rules_to_path_rules(rules);
    let text_rules = rules_to_text_rules(rules);
    let ast_entrypoints = ast_entrypoint_map(rules);

    match check_file(
        &path_rules,
        &text_rules,
        &ast_entrypoints,
        path,
        file,
        settings,
        FixMode::Generate,
        UnsafeFixes::Hint,
    ) {
        Ok(violations) => {
            if violations.messages.is_empty() {
                return String::new();
            }

            let mut output = Vec::new();

            TextEmitter::default()
                .with_show_fix_status(true)
                .with_show_fix_diff(true)
                .with_show_source(true)
                .with_unsafe_fixes(crate::settings::UnsafeFixes::Enabled)
                .emit(&mut output, &violations.messages)
                .unwrap();

            String::from_utf8(output).unwrap()
        }
        Err(msg) => {
            panic!("Failed to process: {msg}");
        }
    }
}
