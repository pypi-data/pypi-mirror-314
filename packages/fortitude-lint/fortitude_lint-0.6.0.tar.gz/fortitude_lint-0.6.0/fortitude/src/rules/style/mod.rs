pub mod double_colon_in_decl;
pub mod end_statements;
pub mod exit_labels;
pub mod line_length;
pub mod old_style_array_literal;
pub mod relational_operators;
pub mod whitespace;

#[cfg(test)]
mod tests {
    use std::convert::AsRef;
    use std::path::Path;

    use anyhow::Result;
    use insta::assert_snapshot;
    use test_case::test_case;

    use crate::registry::Rule;
    use crate::settings::Settings;
    use crate::test::test_path;

    #[test_case(Rule::LineTooLong, Path::new("S001.f90"))]
    #[test_case(Rule::MissingExitOrCycleLabel, Path::new("S021.f90"))]
    #[test_case(Rule::OldStyleArrayLiteral, Path::new("S041.f90"))]
    #[test_case(Rule::DeprecatedRelationalOperator, Path::new("S051.f90"))]
    #[test_case(Rule::UnnamedEndStatement, Path::new("S061.f90"))]
    #[test_case(Rule::MissingDoubleColon, Path::new("S071.f90"))]
    #[test_case(Rule::TrailingWhitespace, Path::new("S101.f90"))]
    #[test_case(Rule::IncorrectSpaceBeforeComment, Path::new("S102.f90"))]
    fn rules(rule_code: Rule, path: &Path) -> Result<()> {
        let snapshot = format!("{}_{}", rule_code.as_ref(), path.to_string_lossy());
        let diagnostics = test_path(
            Path::new("style").join(path).as_path(),
            &[rule_code],
            &Settings::default(),
        )?;
        assert_snapshot!(snapshot, diagnostics);
        Ok(())
    }

    #[test_case(Rule::LineTooLong, Path::new("S001_line_length_20.f90"))]
    fn line_too_long_line_length_20(rule_code: Rule, path: &Path) -> Result<()> {
        let snapshot = format!("{}_{}", rule_code.as_ref(), path.to_string_lossy());
        #[allow(clippy::needless_update)]
        let settings = Settings {
            line_length: 20,
            ..Settings::default()
        };
        let diagnostics = test_path(
            Path::new("style").join(path).as_path(),
            &[rule_code],
            &settings,
        )?;
        assert_snapshot!(snapshot, diagnostics);
        Ok(())
    }
}
