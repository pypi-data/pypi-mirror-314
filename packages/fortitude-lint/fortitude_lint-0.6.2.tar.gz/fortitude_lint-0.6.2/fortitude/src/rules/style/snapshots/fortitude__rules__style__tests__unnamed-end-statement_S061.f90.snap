---
source: fortitude/src/rules/style/mod.rs
expression: diagnostics
snapshot_kind: text
---
./resources/test/fixtures/style/S061.f90:6:3: S061 [*] end statement should be named.
  |
4 |     integer :: x
5 |     ! catch this
6 |   end type
  |   ^^^^^^^^ S061
7 | contains
8 |   subroutine mysub1()
9 |     write (*,*) 'hello world'
  |
  = help: Write as 'end type mytype'.

ℹ Safe fix
3 3 |   type mytype
4 4 |     integer :: x
5 5 |     ! catch this
6   |-  end type
  6 |+  end type mytype
7 7 | contains
8 8 |   subroutine mysub1()
9 9 |     write (*,*) 'hello world'

./resources/test/fixtures/style/S061.f90:11:3: S061 [*] end statement should be named.
   |
 9 |     write (*,*) 'hello world'
10 |     ! catch this
11 |   end subroutine
   |   ^^^^^^^^^^^^^^ S061
12 |   subroutine mysub2()
13 |     write (*,*) 'hello world'
14 |     ! ignore this
   |
   = help: Write as 'end subroutine mysub1'.

ℹ Safe fix
8  8  |   subroutine mysub1()
9  9  |     write (*,*) 'hello world'
10 10 |     ! catch this
11    |-  end subroutine
   11 |+  end subroutine mysub1
12 12 |   subroutine mysub2()
13 13 |     write (*,*) 'hello world'
14 14 |     ! ignore this

./resources/test/fixtures/style/S061.f90:17:1: S061 [*] end statement should be named.
   |
15 |   end subroutine mysub2
16 |   ! catch this
17 | end
   | ^^^ S061
18 | module mymod2
19 |   implicit none
20 |   type mytype
   |
   = help: Write as 'end module mymod1'.

ℹ Safe fix
14 14 |     ! ignore this
15 15 |   end subroutine mysub2
16 16 |   ! catch this
17    |-end
   17 |+end module mymod1
18 18 | module mymod2
19 19 |   implicit none
20 20 |   type mytype

./resources/test/fixtures/style/S061.f90:28:3: S061 [*] end statement should be named.
   |
26 |     myfunc1 = 1
27 |     ! catch this
28 |   end function
   |   ^^^^^^^^^^^^ S061
29 |   integer function myfunc2()
30 |     myfunc2 = 1
31 |     ! ignore this
   |
   = help: Write as 'end function myfunc1'.

ℹ Safe fix
25 25 |   integer function myfunc1()
26 26 |     myfunc1 = 1
27 27 |     ! catch this
28    |-  end function
   28 |+  end function myfunc1
29 29 |   integer function myfunc2()
30 30 |     myfunc2 = 1
31 31 |     ! ignore this

./resources/test/fixtures/style/S061.f90:34:1: S061 [*] end statement should be named.
   |
32 |   end function myfunc2
33 |   ! catch this
34 | end module
   | ^^^^^^^^^^ S061
35 | module mymod3
36 |   interface
37 |     module function foo() result(x)
   |
   = help: Write as 'end module mymod2'.

ℹ Safe fix
31 31 |     ! ignore this
32 32 |   end function myfunc2
33 33 |   ! catch this
34    |-end module
   34 |+end module mymod2
35 35 | module mymod3
36 36 |   interface
37 37 |     module function foo() result(x)

./resources/test/fixtures/style/S061.f90:56:3: S061 [*] end statement should be named.
   |
54 |     x = 1
55 |     ! catch this
56 |   end procedure
   |   ^^^^^^^^^^^^^ S061
57 |   ! catch this
58 | end
59 | submodule (mymod3) mysub2
   |
   = help: Write as 'end procedure foo'.

ℹ Safe fix
53 53 |   module procedure foo
54 54 |     x = 1
55 55 |     ! catch this
56    |-  end procedure
   56 |+  end procedure foo
57 57 |   ! catch this
58 58 | end
59 59 | submodule (mymod3) mysub2

./resources/test/fixtures/style/S061.f90:58:1: S061 [*] end statement should be named.
   |
56 |   end procedure
57 |   ! catch this
58 | end
   | ^^^ S061
59 | submodule (mymod3) mysub2
60 | contains
61 |   module procedure bar
   |
   = help: Write as 'end submodule mysub1'.

ℹ Safe fix
55 55 |     ! catch this
56 56 |   end procedure
57 57 |   ! catch this
58    |-end
   58 |+end submodule mysub1
59 59 | submodule (mymod3) mysub2
60 60 | contains
61 61 |   module procedure bar

./resources/test/fixtures/style/S061.f90:66:1: S061 [*] end statement should be named.
   |
64 |   end procedure bar
65 |   ! catch this
66 | end submodule
   | ^^^^^^^^^^^^^ S061
67 | submodule (mymod3) mysub3
68 | contains
69 |   module procedure baz
   |
   = help: Write as 'end submodule mysub2'.

ℹ Safe fix
63 63 |     ! ignore this
64 64 |   end procedure bar
65 65 |   ! catch this
66    |-end submodule
   66 |+end submodule mysub2
67 67 | submodule (mymod3) mysub3
68 68 | contains
69 69 |   module procedure baz

./resources/test/fixtures/style/S061.f90:79:1: S061 [*] end statement should be named.
   |
77 |   write (*,*) 'hello world'
78 |   ! catch this
79 | end
   | ^^^ S061
   |
   = help: Write as 'end program myprog'.

ℹ Safe fix
76 76 |   implicit none
77 77 |   write (*,*) 'hello world'
78 78 |   ! catch this
79    |-end
   79 |+end program myprog
